src/native/client_bootstrap.c

```diff
@@ -77,34 +77,28 @@ jlong JNICALL Java_software_amazon_awssdk_crt_io_ClientBootstrap_clientBootstrap
     struct aws_host_resolver *resolver = (struct aws_host_resolver *)jni_hr;
 
     if (!elg) {
-        aws_jni_throw_runtime_exception(env, "ClientBootstrap.client_bootstrap_new: Invalid EventLoopGroup");
+        aws_jni_throw_illegal_argument_exception(env, "ClientBootstrap.client_bootstrap_new: Invalid EventLoopGroup");
         return (jlong)NULL;
     }
 
     if (!resolver) {
-        aws_jni_throw_runtime_exception(env, "ClientBootstrap.client_bootstrap_new: Invalid HostResolver");
+        aws_jni_throw_illegal_argument_exception(env, "ClientBootstrap.client_bootstrap_new: Invalid HostResolver");
         return (jlong)NULL;
     }
 
     struct aws_allocator *allocator = aws_jni_get_allocator();
 
     struct shutdown_callback_data *callback_data = aws_mem_calloc(allocator, 1, sizeof(struct shutdown_callback_data));
     if (!callback_data) {
-        aws_jni_throw_runtime_exception(env, "ClientBootstrap.client_bootstrap_new: Unable to allocate");
+        aws_jni_throw_last_error(env, "ClientBootstrap.client_bootstrap_new: Unable to allocate");
```

Comment:
If we're going to do this, might as well add " shutdown callback data" to the end of the string